fili-sql/src/main/java/com/yahoo/bard/webservice/sql/presto/DruidQueryToPrestoConverter.java

```diff
@@ -0,0 +1,296 @@
+    }
+
+    /**
+     * Gets all the dimensions from a druid query as fields for calcite.
+     *
+     * @param builder  The RelBuilder created with Calcite.
+     * @param druidQuery  The query to find the having filter from.
+     * @param apiToFieldMapper  The mapping from api to physical name.
+     *
+     * @return the list of dimensions as {@link RexNode} for Calcite's builder.
+     */
+    private List<RexNode> getDimensionFields(
+            RelBuilder builder,
+            DruidAggregationQuery<?> druidQuery,
+            ApiToFieldMapper apiToFieldMapper
+    ) {
+        return druidQuery.getDimensions().stream()
+                .map(Dimension::getApiName)
+                .map(apiToFieldMapper)
+                .map(builder::field)
+                .collect(Collectors.toList());
+    }
+
+
+    /**
+     * Find all druid aggregations and convert them to {@link org.apache.calcite.tools.RelBuilder.AggCall}.
+     * Difference between this and the base class is the special column lower underscore to lower camel case conversion
+     * for Druid column names
+     * @param builder  The RelBuilder created with Calcite.
+     * @param druidQuery  The druid query to get the aggregations of.
+     * @param apiToFieldMapper  The mapping from api to physical name.
+     *
+     * @return the list of aggregations.
+     */
+    @Override
+    protected List<RelBuilder.AggCall> getAllQueryAggregations(
+            RelBuilder builder,
+            DruidAggregationQuery<?> druidQuery,
+            ApiToFieldMapper apiToFieldMapper
+    ) {
+        return druidQuery.getAggregations()
+                .stream()
+                .map(aggregation -> druidSqlAggregationConverter.apply(aggregation, apiToFieldMapper))
+                .map(optionalSqlAggregation -> optionalSqlAggregation.orElseThrow(() -> {
+                    String msg = "Couldn't build sql aggregation with " + optionalSqlAggregation;
+                    LOG.debug(msg);
+                    return new RuntimeException(msg);
+                }))
+                .map(sqlAggregation -> builder.aggregateCall(
+                        sqlAggregation.getSqlAggFunction(),
+                        false,
+                        null,
+                        CaseFormat.LOWER_UNDERSCORE.to(
+                                CaseFormat.LOWER_CAMEL,
+                                sqlAggregation.getSqlAggregationAsName()
+                        ),
+                        builder.field(sqlAggregation.getSqlAggregationFieldName())
+                ))
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Returns the post-aggregations of the query.
+     *
+     * @param builder the RelBuilder
+     * @param druidQuery the source druid query
+     * @param apiToFieldMapper api column to logic column name mapping
+     * @return a list of RexNode representing the post aggregation
+     */
+    private List<RexNode> getPostAggregations(
```

Comment:
This should be pulled up along with `DruidPostAggregationToPresto`

Reply:
Discuss

Reply:
Fixed

Reply:
Fixed